src/backend/base/langflow/services/tracing/arize_phoenix.py

```diff
@@ -0,0 +1,323 @@
+        if processed_inputs:
+            child_span.set_attribute(SpanAttributes.INPUT_VALUE, self._safe_json_dumps(processed_inputs))
+            child_span.set_attribute(SpanAttributes.INPUT_MIME_TYPE, OpenInferenceMimeTypeValues.JSON.value)
+
+        processed_metadata = self._convert_to_arize_phoenix_types(metadata) if metadata else {}
+        if processed_metadata:
+            for key, value in processed_metadata.items():
+                child_span.set_attribute(f"{SpanAttributes.METADATA}.{key}", value)
+
+        self.child_spans[trace_id] = child_span
+
+    @override
+    def end_trace(
+        self,
+        trace_id: str,
+        trace_name: str,
+        outputs: dict[str, Any] | None = None,
+        error: Exception | None = None,
+        logs: Sequence[Log | dict] = (),
+    ) -> None:
+        """Ends a trace span, attaching outputs, errors, and logs as attributes."""
+        if not self._ready or trace_id not in self.child_spans:
+            return
+
+        child_span = self.child_spans[trace_id]
+
+        processed_outputs = self._convert_to_arize_phoenix_types(outputs) if outputs else {}
+        if processed_outputs:
+            child_span.set_attribute(SpanAttributes.OUTPUT_VALUE, self._safe_json_dumps(processed_outputs))
+            child_span.set_attribute(SpanAttributes.OUTPUT_MIME_TYPE, OpenInferenceMimeTypeValues.JSON.value)
+
+        logs_dicts = [log if isinstance(log, dict) else log.model_dump() for log in logs]
+        processed_logs = (
+            self._convert_to_arize_phoenix_types({log.get("name"): log for log in logs_dicts}) if logs else {}
+        )
+        if processed_logs:
+            for key, value in processed_logs.items():
+                child_span.set_attribute(f"logs.{key}", value)
+
+        if error:
+            error_string = self._error_to_string(error)
+            child_span.set_status(Status(StatusCode.ERROR, error_string))
+            child_span.set_attribute("error.message", error_string)
+
+            if isinstance(error, Exception):
+                child_span.record_exception(error)
+            else:
+                exception_type = error.__class__.__name__
+                exception_message = str(error)
+                if not exception_message:
+                    exception_message = repr(error)
+                attributes: dict[str, AttributeValue] = {
+                    OTELSpanAttributes.EXCEPTION_TYPE: exception_type,
+                    OTELSpanAttributes.EXCEPTION_MESSAGE: exception_message,
+                    OTELSpanAttributes.EXCEPTION_ESCAPED: False,
+                    OTELSpanAttributes.EXCEPTION_STACKTRACE: error_string,
+                }
+                child_span.add_event(name="exception", attributes=attributes)
+        else:
+            child_span.set_status(Status(StatusCode.OK))
+
+        child_span.end(end_time=self._get_current_timestamp())
+        self.child_spans.pop(trace_id)
```

Comment:
```suggestion
        self.child_spans.pop(UUID(trace_id))
```
mypy requires a conversion to match its type definition.